### ISSUE TITLE
Blur event is not triggering for checkbox 

### ISSUE THREAD

**Issue by Janiece19** [ngx-formly/ngx-formly#2602]
**Description**
Binding Blur event to a checkbox field in angular formly, it is not triggering event on checkbox click

**Minimal Reproduction**

https://stackblitz.com/edit/angular-2fvxks?file=src/app/app.component.ts

**Your Environment**

- Angular version: 9.1.2
- Formly version: 5.5.15

**Comment by aitboudad** [comment]
could you please check if the issue exists with our latest version `v5.10.6` also which UI you're using?

**Comment by Janiece19** [comment]
I am using angular material.

https://stackblitz.com/edit/angular-2fvxks?file=src/app/app.component.ts

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v5.10.7](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.10.7) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!

**Comment by peterk7** [comment]
Hi @aitboudad, 
My project auto-updated to v5.10.7 and a page that contained the boolean type (checkbox) started causing issues and console log errors. (v5.10.6 works fine for us)

We debugged it to this thread and this update.

The problem that we found is with this code (checkbox.type.ts):
`ngOnDestroy() {
    this.focusMonitor.stopMonitoring(this.checkbox._inputElement);
  }`

When updating the form, the resolveFieldTypeRef function gets called (fromly.config.ts) and it calls destroy for every field component:
`componentRef.destroy();`

Given the checkbox didnt render yet, `this.checkbox === undefined` and throws a log error: "Cannot read property '_inputElement' of undefined".

Maybe an error check should be added there for that case?

Thanks.
